zipline/testing/fixtures.py

```diff
@@ -315,72 +333,218 @@ def init_class_fixtures(cls):
         cls.trading_days = all_days[start_loc:end_loc + 1]
 
 
-class WithTempdir(object):
+class WithTmpDir(object):
     """
-    ZiplineTestCase mixin providing cls.tempdir as a class-level fixture.
+    ZiplineTestCase mixing providing cls.tmpdir as a class-level fixture.
 
-    After init_class_fixtures has been called, `cls.tempdir` is populated
-    with a TempDirectory object.
+    After init_class_fixtures has been called, `cls.tmpdir` is populated with
+    a `testfixtures.TempDirectory` object whose path is `cls.TMP_DIR_PATH`.
 
-    The default value of TEMPDIR_PATH is None.
-    Inheritors can override TEMPDIR_PATH to path argument to `TempDirectory`
+    The default value of TMP_DIR_PATH is None which will create a unique
+    directory in /tmp.
     """
-
-    TEMPDIR_PATH = None
+    TMP_DIR_PATH = None
 
     @classmethod
     def init_class_fixtures(cls):
-        super(WithTempdir, cls).init_class_fixtures()
+        super(WithTmpDir, cls).init_class_fixtures()
+        cls.tmpdir = cls.enter_class_context(tmp_dir(cls.TMP_DIR_PATH))
 
-        cls.tempdir = TempDirectory(path=cls.TEMPDIR_PATH)
 
-        cls.add_class_callback(cls.tempdir.cleanup)
+class WithInstanceTmpDir(object):
```

Comment:
Should we give the fixture generated here a different name so we don't collide if you use both?  I could imagine a test that needs a daily bar reader and also needs a separate per-test tempdir (though maybe not a whole suite of them, in which case maybe we should just make it an error to use both and say you should use TempDirectory directly if you already have the class-level one.)